phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -883,6 +903,14 @@ public void readFields(DataInput input) throws IOException {
         dataEmptyKeyValueCF = Bytes.readByteArray(input);
         emptyKeyValueCFPtr = new ImmutableBytesPtr(Bytes.readByteArray(input));
         
+        int numIndexedExpressions = WritableUtils.readVInt(input);
+        indexedExpressions = Sets.newLinkedHashSetWithExpectedSize(numIndexedExpressions);        
+        for (int i = 0; i < numIndexedExpressions; i++) {
```

Comment:
Does it help if you treat them the same, as a regular column is an expression if you parse it?